3.preprocessing_features/scripts/1.sc_quality_control.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Perform single-cell quality control
+# 
+# In this notebook, we perform single-cell quality control using coSMicQC. We filter the single cells by identifying outliers with z-scores, and use either combinations of features or one feature for each condition. We use features from the AreaShape and Intensity modules to assess the quality of the segmented single-cells:
+# 
+# ### Assessing poor nuclei segmentation
+# 
+# Due to high confluence, sometimes nuclei overlap on top of each other, creating highly intense clusters within the Hoechst channel. To identify these nuclei, we use:
+# 
+# - **Nuclei Area:** This metric quantifies the number of pixels in a nucleus segmentation. 
+# We detect nuclei that are abnormally large, which likely indicates poor nucleus segmentation where overlapping nuclei are merged into one segmentation. 
+# - **Nuclei Intensity:** This metric quantifies the total intensity of all pixels in a nucleus segmentation. 
+#   
+# In combination with abnormally large nuclei, we detect nuclei that are also highly intense, likely indicating that this a group of overlapped nuclei.
+# 
+# `We utilize the same thresholds for this section as set in the [cellpainting_predicts_cardiac_fibroblasts](https://github.com/WayScience/cellpainting_predicts_cardiac_fibrosis) repository.`
+# 
+# We decided in this notebook to also include finding nuclei with very **LOW** intensity which more than likely correlates with mis-segmented nuclei from the background. 
+# These occur due to very cytotoxic compounds that kill all the cells, leaving empty FOVs where the segmentation parameters sometimes decides to segment nuclei from nothing.
+# 
+# `We determine our own threshold using the first batch of data (4 plates from layout one) that we will use for the rest of the plates.`
+# 
+# ### Assessing poor cell segmentation
+# 
+# Also due to high confluence, images with large, intense clusters of cells leads to errors in the segmentation algorithm that causes cells around the cluster to segmented incorrectly. 
+# When this happens, a cell is segmented around the same segmentation as the nucleus, giving it the same area which is very small for a normal cardiac fibroblast cell. To detect poorly segmented cells, we use:
+# 
+# - **Cells area:** The cells Area metric quantifies the number of pixels in a cell segmentation.
+# 
+# `We update the threshold for this dataset from the original thresholds since we determined that there were too many good cells being removed. We update the threshold to be more loose.`
```

Comment:
Love the documentation!

Reply:
Thank you! I hope it made sense all the quality control stuff lol. I need to include the image data at some point in here.